src/Core/CacheManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Core
+{
+    public class CacheManager
```

Comment:
I'm thinking this could be named better to say what it is. And maybe also have a more generic implementation. Like RefreshingValue or something like that.

And signature for it could be:
```csharp
class RefreshingValue<T>
    ctor(Func<Task<T>> valueFactory)
    T Value { get; }
```

Reply:
I followed your suggestion and created something called AsyncPollingCache<T>. I also provide a DictionaryPollingCache : AsyncPollingCache<IReadOnlyDictiona........> as an easy default to use for our current traduora use case.

Do you think it's better now?